Log Olm session identifiers

Anderas opened the pull request:
Log Olm session identifier when encrypting or decrypting messages, or when creating a new session from pre-key message

jplatte commented on crates/matrix-sdk-crypto/src/olm/account.rs:
```diff
@@ -293,59 +293,77 @@ impl Account {
+            SessionType::Existing(s) => s.session_id(),
+        };
+        trace!(
+            ?sender_key,
+            "Successfully decrypted an Olm message with Olm session `{}`",
+            session_id
```
```suggestion
            ?sender_key, ?session_id,
            "Successfully decrypted an Olm message",
```

jplatte reviewed (commented):
I think we should use tracing fields like elsewhere, I've added an example below.

poljar commented:
I added some new spans and moved the common fields we want to record into the span. A small vodozemac change helped us to record the session id sooner as well: https://github.com/matrix-org/vodozemac/pull/99.

Anderas commented:
> I added some new spans and moved the common fields we want to record into the span. A small vodozemac change helped us to record the session id sooner as well: [matrix-org/vodozemac#99](https://github.com/matrix-org/vodozemac/pull/99).

Looks great, thanks
